catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -0,0 +1,396 @@
+        registryMap.put("Collections", collectionMap);
+
+        Map<String, Object> serviceMap = getServiceInfo(registryPackage.getRegistryObjectList());
+        registryMap.put("Services", serviceMap);
+
+        Map<String, Object> organizationMap = getOrganizationInfo(registryPackage);
+        registryMap.put("Organizations", organizationMap);
+
+        Map<String, Object> contactMap = getContactInfo(registryPackage);
+        registryMap.put("Contacts", contactMap);
+
+        return registryMap;
+    }
+
+    private Map<String, Object> getServiceInfo(RegistryObjectListType registryObjectListType) {
+        Map<String, Object> serviceInfo = new HashMap<>();
+        for (JAXBElement id : registryObjectListType.getIdentifiable()) {
+            RegistryObjectType registryObjectType = (RegistryObjectType) id.getValue();
+            if (registryObjectType instanceof ServiceType
+                    && RegistryConstants.REGISTRY_SERVICE_OBJECT_TYPE.equals(registryObjectType.getObjectType())) {
+                ServiceType service = (ServiceType) registryObjectType;
+
+                String serviceName = "Service";
+                if (service.isSetName()) {
+                    serviceName = RegistryPackageUtils.getStringFromIST(service.getName());
+                }
+                Map<String, Object> serviceProperties = new HashMap<>();
+                for (ServiceBindingType binding : service.getServiceBinding()) {
+                    Map<String, Object> bindingProperties;
+                    String bindingType;
+                    bindingProperties = getCustomSlots(binding.getSlot());
+                    if (!bindingProperties.containsKey(BINDING_TYPE) || bindingProperties.get(
+                            BINDING_TYPE) == null) {
+                        continue;
+                    }
+                    bindingType = bindingProperties.get(BINDING_TYPE)
+                            .toString();
+                    bindingProperties.remove(BINDING_TYPE);
+                    serviceProperties.put(bindingType, bindingProperties);
+                }
+                serviceInfo.put(serviceName, serviceProperties);
+            }
+        }
+        return serviceInfo;
+    }
+
+    private Map<String, Object> getExtrinsicInfo(RegistryPackageType registryPackage) {
+        Map<String, Object> generalMap = new HashMap<>();
+        Map<String, Object> collectionMap = new HashMap<>();
+
+        List<ExtrinsicObjectType> extrinsicObjectTypes = RegistryPackageUtils.getExtrinsicObjects(
+                registryPackage);
+        for (ExtrinsicObjectType extrinsicObjectType : extrinsicObjectTypes) {
+            if (extrinsicObjectType.isSetObjectType()) {
+                if (extrinsicObjectType.getObjectType()
+                        .equals(RegistryConstants.REGISTRY_NODE_OBJECT_TYPE)) {
+                    if (extrinsicObjectType.isSetName()) {
+                        registryName =
```

Comment:
I am not sure how jaxrs works, but I think if you can get two concurrent GET requests, one could clobber the other and the `registryName` variable would be incorrect for one of the requests.